src/ir/type-updating.cpp

```diff
@@ -146,17 +147,38 @@ GlobalTypeRewriter::TypeMap GlobalTypeRewriter::rebuildTypes(
 #endif
   auto& newTypes = *buildResults;
 
-  // Map the old types to the new ones.
+  // Map the old types to the new ones, and check if anything changed.
+  bool changed = false;
   TypeMap oldToNewTypes;
   for (auto [type, index] : typeIndices) {
-    oldToNewTypes[type] = newTypes[index];
+    auto newType = newTypes[index];
+    oldToNewTypes[type] = newType;
+    if (newType != type) {
```

Comment:
If you want to be fancy about this ;)
```suggestion
    if (type != (oldToNewTypes[type] = newTypes[index])) {
```

Reply:
Moot after the last commit, but that would definitely have been more concise.